File: app/src/main/java/com/widiarifki/screentest/presentation/menu/MenuPresenter.java
package com.widiarifki.screentest.presentation.menu;

/**
 * Created by widiarifki on 27/02/2018.
 */

public class MenuPresenter {
    IMenuView mView;
    String mEventName;
    String mGuestName;

    public MenuPresenter(IMenuView view) {
        mView = view;
    }

    public void setEventName(String eventName) {
        mEventName = eventName;
        mView.writeEvent(mEventName);
    }

    public void setGuestName(String guestName) {
        mGuestName = guestName;
        mView.writeGuest(mGuestName);
    }
}

File: app/src/main/java/com/widiarifki/screentest/presentation/menu/MenuPresenter.kt
package com.widiarifki.screentest.presentation.menu

import com.widiarifki.screentest.model.User
import io.realm.Realm

/**
 * Created by widiarifki on 27/02/2018.
 */

class MenuPresenter(internal var mView: IMenuView) {

    fun getUsername() : String?{
        val realm = Realm.getDefaultInstance()
        val realmQuery = realm.where(User::class.java)
        val user: User? = realmQuery.findFirst()
        return user?.name
    }
}
